lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/util/TrinoJsonFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.util;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonFactoryBuilder;
+import com.fasterxml.jackson.core.StreamReadConstraints;
+import org.gaul.modernizer_maven_annotations.SuppressModernizer;
+
+public final class TrinoJsonFactory
+{
+    private TrinoJsonFactory() {}
+
+    @SuppressModernizer
+    public static JsonFactoryBuilder trinoJsonFactoryBuilder()
+    {
+        return trinoJsonFactoryBuilder(new JsonFactory());
+    }
+
+    @SuppressModernizer
+    public static JsonFactoryBuilder trinoJsonFactoryBuilder(JsonFactory factory)
+    {
+        return new JsonFactoryBuilder(factory)
+                .streamReadConstraints(StreamReadConstraints.builder()
+                        .maxStringLength(Integer.MAX_VALUE)
+                        .maxNestingDepth(Integer.MAX_VALUE)
+                        .maxNumberLength(Integer.MAX_VALUE)
+                        .build());
+    }
+
+    public static JsonFactory trinoJsonFactory()
```

Comment:
move to `JsonUtils` in plugin toolkit.

we do not need two different util classes for json, do we?